stacktrace.go

```diff
@@ -52,35 +53,63 @@ type StacktraceFrame struct {
 	InApp        bool     `json:"in_app"`
 }
 
-// GetOrNewStacktrace tries to get stacktrace from err as an interface of github.com/pkg/errors, or else NewStacktrace()
-func GetOrNewStacktrace(err error, skip int, context int, appPackagePrefixes []string) *Stacktrace {
-	type stackTracer interface {
-		StackTrace() []runtime.Frame
-	}
-	stacktrace, ok := err.(stackTracer)
-	if !ok {
-		return NewStacktrace(skip+1, context, appPackagePrefixes)
-	}
+// extractFramesFromPcs translates slice of stack trace pointers into usable frames
+func extractFramesFromPcs(pcs []uintptr, context int, appPackagePrefixes []string) []*StacktraceFrame {
 	var frames []*StacktraceFrame
-	for f := range stacktrace.StackTrace() {
-		pc := uintptr(f) - 1
-		fn := runtime.FuncForPC(pc)
-		var fName string
-		var file string
-		var line int
-		if fn != nil {
-			file, line = fn.FileLine(pc)
-			fName = fn.Name()
-		} else {
-			file = "unknown"
-			fName = "unknown"
-		}
-		frame := NewStacktraceFrame(pc, fName, file, line, context, appPackagePrefixes)
+	callersFrames := runtime.CallersFrames(pcs)
+
+	for {
+		fr, more := callersFrames.Next()
+		frame := NewStacktraceFrame(fr.PC, fr.Function, fr.File, fr.Line, context, appPackagePrefixes)
 		if frame != nil {
 			frames = append([]*StacktraceFrame{frame}, frames...)
 		}
+		if !more {
+			break
+		}
+	}
+
+	return frames
+}
+
+// GetOrNewStacktrace tries to get stacktrace from err as an interface of github.com/pkg/errors, or else NewStacktrace()
+// Use of reflection allows us to not have a hard dependency on any given package, so we don't have to import it
+func GetOrNewStacktrace(err error, skip int, context int, appPackagePrefixes []string) *Stacktrace {
```

Comment:
This is amazing! Is it possible though to have the Client have a configurable GetStackTrace method so that anyone can pas in the error implementation?

Reply:
I want to make it happen in the sentry-go